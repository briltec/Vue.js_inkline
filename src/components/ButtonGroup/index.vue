<template src="./template.html" />
<script src="./script.js" />
<style src="../Button/style.styl" lang="stylus" scoped />
<style src="./style.styl" lang="stylus" scoped />
