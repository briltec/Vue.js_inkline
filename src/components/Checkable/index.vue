<script src="./script.js" />
<!--<style src="./style.styl" lang="stylus" />-->
