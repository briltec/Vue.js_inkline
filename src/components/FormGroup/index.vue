<template src="template.html" />
<script src="script.js" />
<style src="../Input/style.styl" lang="stylus" scoped />
<style src="style.styl" lang="stylus" scoped />
