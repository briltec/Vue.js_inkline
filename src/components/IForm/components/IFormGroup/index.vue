<template src="./template.html" />
<style src="./style.scss" lang="scss" />
<script src="./script.js" />
