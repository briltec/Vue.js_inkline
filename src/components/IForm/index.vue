<template src="./template.html" />
<script src="./script.ts" lang="ts" />
<style src="../IInput/style.scss" lang="scss" scoped />
<style src="./style.scss" lang="scss" scoped />
