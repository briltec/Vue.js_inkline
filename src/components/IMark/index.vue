<template src="./template.html" />
<script src="./script.ts" lang="ts" />
