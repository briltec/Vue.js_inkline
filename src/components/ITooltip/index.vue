<template src="./template.html" />
<script src="./script.ts" lang="ts" />
<style src="./style.scss" lang="scss" />
