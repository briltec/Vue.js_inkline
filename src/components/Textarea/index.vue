<template src="./template.html" />
<style src="../Input/style.styl" lang="stylus" />
<script src="./script.js" />
