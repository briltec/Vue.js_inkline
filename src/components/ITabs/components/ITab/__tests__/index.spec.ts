import { render } from '@testing-library/vue';
import { ITab } from '@inkline/inkline/components';

describe('Components', () => {
    describe('ITab', () => {
        const props = {
            name: 'tab'
        };

        it('should be named correctly', () => {
            expect(ITab.name).toEqual('ITab');
        });

        it('should render correctly', () => {
            const wrapper = render(ITab, { props });

            expect(wrapper.html()).toMatchSnapshot();
        });

        describe('props', () => {
            describe('name', () => {
                it('should have randomly generated name uid', () => {
                    const wrapper = render(ITab);

                    expect(wrapper.container.firstChild).toHaveAttribute('name', expect.stringContaining('tab'));
                });
            });
        });

        describe('computed', () => {
            describe('classes', () => {
                it('should have \'-active\' class if active', () => {
                    const wrapper = render(ITab, {
                        global: {
                            provide: {
                                tabs: {
                                    active: 'tab'
                                }
                            }
                        },
                        props
                    });

                    expect(wrapper.container.firstChild).toHaveClass(
                        '-active'
                    );
                });

                it('should not have \'-active\' class if not active', () => {
                    const wrapper = render(ITab, {
                        global: {
                            provide: {
                                tabs: {
                                    active: 'other'
                                }
                            }
                        },
                        props
                    });

                    expect(wrapper.container.firstChild).not.toHaveClass(
                        '-active'
                    );
                });
            });
        });
    });
});
