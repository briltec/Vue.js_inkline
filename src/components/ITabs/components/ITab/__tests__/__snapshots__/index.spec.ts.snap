// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components ITab should render correctly 1`] = `
<div class="tab" role="tabpanel" name="tab" aria-hidden="true" aria-labelledby="tab-heading-tab" style="display: none;">
  <div class="tab-body"></div>
</div>
`;
